lib/symlink/link-file.js

```diff
@@ -22,25 +24,41 @@ function linkStream(optResolver) {
     //    For example, JetBrains product lines will delete the entire
     //    contents of the TARGET directory because the product does not
     //    realize it's a symlink as the JVM and Node return false for isSymlink.
-    var useJunctions = optResolver.resolve('useJunctions', file);
+    var useJunctions = isWindows && optResolver.resolve('useJunctions', file);
 
-    var symDirType =  useJunctions ? 'junction' : 'dir';
-    var symType = isDirectory ? symDirType : 'file';
     var isRelative = optResolver.resolve('relativeSymlinks', file);
+    var flag = optResolver.resolve('flag', file);
 
-    // This is done after prepare() to use the adjusted file.base property
-    if (isRelative && symType !== 'junction') {
-      file.symlink = path.relative(file.base, file.symlink);
+    if (isWindows) {
+      fs.stat(file.symlink, onStat);
+    } else {
+      onStat(null, file.stat);
     }
 
-    var flag = optResolver.resolve('flag', file);
+    function onStat(statErr, stat) {
+      if (statErr && statErr.code !== 'ENOENT') {
+        return onWritten(statErr);
+      }
+      if (!stat) {
+        // If target doesn't exist, let's look at the vinyl to see which kind
+        // of dangling link we should to create...
+        stat = file.stat;
```

Comment:
I think this is proper here but not in the `.dest()` for symlinks.

Reply:
Not sure I understand why it ought to be different?

Reply:
In the case someone does `vfs.src('my/cool/directory').pipe(vfs.symlink('my/link/location'))` - the `file.stat` property would be that of the thing (directory or file) we are creating a link for and we can rely on the stats.  The only case where that wouldn't work would be a symlink for a symlink (which would have to be dangling to not be handled by the code above this). 

This is opposite of `.dest()` where we will **only** end up in the symlink branch if the Vinyl is already symbolic.